app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java

```diff
@@ -312,10 +314,10 @@ public void onRequestPermissionsResult(int requestCode,
             return;
 
         switch (requestCode) {
-            case PERMISSION_REQUEST_WRITE_EXTERNAL_STORAGE_RESULT: {
+            case REQUEST_PERMISSIONS_CODE: {
                 if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
-                    if (mOpenSelectImages)
-                        selectImages();
+                    mPermissionGranted = true;
+                    selectImages();
```

Comment:
you have removed the if condition for this case
`if (mOpenSelectImages)`

Reply:
yes I have because it's of no use.
earlier we were getting permission from user at two places:
* when we enter ImagesToPdfFragment
* when user click on add images
and this is why mOpenSelectImages was used.
I have removed the code to get permission on  entering ImagesToPdfFragment becuase it's useless, only when user click on add images, he will be asked for permission. so as now we are requesting permission at only one place, mOpenSelectImages  is not required

Reply:
Okay. Sounds good :)